src/Integration/Suppression/SonarQubeIssuesProvider.cs

```diff
@@ -55,36 +55,25 @@ public sealed class SonarQubeIssuesProvider : ISonarQubeIssuesProvider
         private CancellationTokenSource cancellationTokenSource;
         private readonly IThreadHandling threadHandling;
 
-        public SonarQubeIssuesProvider(ISonarQubeService sonarQubeService, string sonarQubeProjectKey, ITimerFactory timerFactory,
-            ILogger logger) : this(sonarQubeService, sonarQubeProjectKey, timerFactory, logger, new ThreadHandling())
+        public SonarQubeIssuesProvider(ISonarQubeService sonarQubeService, string sonarQubeProjectKey,
+            ILogger logger) : this(sonarQubeService, sonarQubeProjectKey, logger, new TimerFactory(), new ThreadHandling())
         {
-
         }
 
-
-        internal SonarQubeIssuesProvider(ISonarQubeService sonarQubeService, string sonarQubeProjectKey, ITimerFactory timerFactory,
-            ILogger logger, IThreadHandling threadHandling)
+        internal /* for testing */ SonarQubeIssuesProvider(ISonarQubeService sonarQubeService, string sonarQubeProjectKey,
+            ILogger logger, ITimerFactory timerFactory, IThreadHandling threadHandling)
         {
-            if (sonarQubeService == null)
-            {
-                throw new ArgumentNullException(nameof(sonarQubeService));
-            }
             if (string.IsNullOrWhiteSpace(sonarQubeProjectKey))
             {
                 throw new ArgumentNullException(nameof(sonarQubeProjectKey));
             }
-            if (timerFactory == null)
-            {
-                throw new ArgumentNullException(nameof(timerFactory));
-            }
-            if (logger == null)
-            {
-                throw new ArgumentNullException(nameof(logger));
-            }
 
-            this.sonarQubeService = sonarQubeService;
+            Debug.Assert(timerFactory != null, "timeFactory should not be null");
+            Debug.Assert(threadHandling != null, "threadHandling should not be null");
```

Comment:
Product code should only ever be calling the `public` constructor, so it will never pass these arguments.

`timerFactory` and `threadHandling` are test-only so asserting that they are not null is enough.